packages/manager/src/features/Billing/PdfGenerator/PdfGenerator.ts

```diff
@@ -99,16 +104,29 @@ const addLeftHeader = (
 
     if (countryTax) {
       addLine(`${countryTax.tax_name}: ${countryTax.tax_id}`);
+
+      const { tax_ids, tax_name } = countryTax;
+      const { EU } = countryTaxNames;
+
+      if (tax_name === EU && tax_ids?.B2B) {
+        const { tax_id: b2bTaxId, tax_name: b2bTaxName } = tax_ids.B2B;
+        addLine(`${b2bTaxName}: ${b2bTaxId}`);
+      }
     }
     /**
-     * M3-7847 Add Akamai's Japanese QI System ID to Japanese Invoices.
+     * [M3-7847, M3-8008] Add Akamai's Japanese QI System ID to Japanese Invoices.
      * Since LD automatically serves Tax data based on the user's
      * we can check on qi_registration field to render QI Registration.
      * */
     if (countryTax && countryTax.qi_registration) {
-      const line = `QI Registration # ${countryTax.qi_registration}`;
-      addLine(line);
+      const qiRegistration = `QI Registration # ${countryTax.qi_registration}`;
+      addLine(qiRegistration);
+    }
+
+    if (countryTax?.tax_name === countryTaxNames.JP && countryTax?.tax_info) {
```

Comment:
LaunchDarkly is setup to conditionally serve tax information based on country. Do we need to perform this hardcoded check?

```suggestion
    if (countryTax?.tax_info) {
```